### PR TITLE
Update helm release workflow

### PR DESCRIPTION
**Pull request by xpuska513**
## What
* Update helm release workflow so it'll follow its own versioning properly using tags
* Fix issue with incorrect version pattern
## How
* Instead of using airbyte oss tags, use own instead(it'll look like `v${{ major }}.${{ minor }}.${{ patch }}-helm` this tag will be pushed on each helm chart release + a release on gh pages will be created
* After this a PR will be created containing updated Chart.yaml files(`version` field will be updated for beautiful README.md and readme on artifacthub)
* This will allow us to keep chart in this repo and also keep own helm versioning

### CODE REVIEW

**Review by cuyk — APPROVED** [review]

**Comment by xpuska513** [comment]
Before merging I'll create a new tag in repo, so versioning will work correctly

**Review by git-phu — APPROVED** [review]
nice!

seems reasonable

File: .github/workflows/publish-helm-charts.yml [hunk 1]
```diff
@@ -30,11 +33,21 @@ jobs:
...
           change_path: "./charts"
           bump_each_commit: true
+          namespace: helm
+
+      - name: "Remove -helm from ver number"
+        shell: bash
+        env:
+          VERSION: ${{ steps.sem-ver.outputs.version }}
+        id: ver
+        run: echo "::set-output name=fragment::${VERSION%%-*}"
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml:43]: I think set output is on it's deathbed... though I might be mistaken

File: .github/workflows/publish-helm-charts.yml [hunk 2]
```diff
@@ -30,11 +33,21 @@ jobs:
           format: "${major}.${minor}.${patch}"
           change_path: "./charts"
           bump_each_commit: true
+          namespace: helm
+
+      - name: "Remove -helm from ver number"
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml:38]: it took me a minute to walk through what this is doing, can we get an example string in the comments like
`1.2.3-helm -> 1.2.3` or something?

File: .github/workflows/publish-helm-charts.yml [hunk 3]
```diff
@@ -51,14 +64,17 @@ jobs:
...
         working-directory: ./airbyte/charts
         run: |
-          sed -i "s/    version: placeholder/    version: ${{ needs.generate-semantic-version.outputs.next-version }}/g" airbyte/Chart.yaml
+          sed -i -E "s/    version: [[:digit:]].[[:digit:]].[[:digit:]]/    version: ${{ needs.generate-semantic-version.outputs.next-version }}/g" airbyte/Chart.yaml
+          sed -i -E 's/version: [0-9]+\.[0-9]+\.[0-9]+/version: ${{ needs.generate-semantic-version.outputs.next-version }}/' airbyte/Chart.yaml
+
 
       - name: "Helm package"
         shell: bash
         run: |
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml:73]:
```suggestion
        run: |
            set -x # or is it set +x for actual verbose stuff, I never remember.
```
I think if anyone has to debug this, more output would help

File: .github/workflows/publish-helm-charts.yml [hunk 4]
```diff
@@ -51,14 +64,17 @@ jobs:
         shell: bash
         working-directory: ./airbyte/charts
         run: |
-          sed -i "s/    version: placeholder/    version: ${{ needs.generate-semantic-version.outputs.next-version }}/g" airbyte/Chart.yaml
+          sed -i -E "s/    version: [[:digit:]].[[:digit:]].[[:digit:]]/    version: ${{ needs.generate-semantic-version.outputs.next-version }}/g" airbyte/Chart.yaml
+          sed -i -E 's/version: [0-9]+\.[0-9]+\.[0-9]+/version: ${{ needs.generate-semantic-version.outputs.next-version }}/' airbyte/Chart.yaml
+
 
       - name: "Helm package"
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml:71]: I know it's not part of the PR but can we make this action more clear?

ie: 
```suggestion
      - name: "Set Helm Package SEMVAR for all charts"
```

File: .github/workflows/publish-helm-charts.yml [hunk 5]
```diff
@@ -89,3 +105,21 @@ jobs:
           add: '.'
           cwd: './airbyte-oss/'
       
+      - name: Commit and push changes
+        uses: EndBug/add-and-commit@v9
+        with:
+          message: 'Bump release to ${{ needs.generate-semantic-version.outputs.tag }}'
+          add: '.'
```
Comment (supertopher) [inline .github/workflows/publish-helm-charts.yml]: do we need to set current working directory to defend against changes to this script that may move current working directory?  

I'm not sure if new actions all start in the same spot or not

**Review by supertopher — COMMENTED** [review]
edit: my comments were too late ¯\_(ツ)_/¯

**Pull request merged**